software/duet_config_files/duet2/rrf2/macros/config_checks.g

```diff
@@ -0,0 +1,73 @@
+; Machine Prechecks
+M84
+M291 R"Machine Prechecks" P"Please move printhead to center of bed, and bed to middle height, and tool changer axis to middle." S3
+
+
+; Endstop Checks
+M291 R"Endstop Checks" P"Walk through each endstop and make sure it's detected and wired properly" S3
+
+; Prechecks
+if sensors.endstops[0].triggered
+  M291 R"X Axis Error" P"X Endstop is already triggered! Either it's depressed, wired wrong, or broken." S3
+if sensors.endstops[1].triggered
+  M291 R"Y Axis Error" P"Y Endstop is already triggered! Either it's depressed, wired wrong, or broken." S3
+if sensors.probes[0].value[0] != 0
+  M291 R"Z Axis Error" P"Z Probe is already triggered! Either it's depressed, wired wrong, or broken." S3
+if sensors.endstops[3].triggered
+  M291 R"U Axis Error" P"U Endstop is already triggered! Either it's depressed, wired wrong, or broken." S3
+
+; X
+echo "Please depress X endstop for 1s"
+M577 X S1
+echo "X endstop test passed"
+
+; Y
+echo "Please depress Y endstop for 1s"
+M577 Y S1
+echo "Y endstop test passed"
+
+; Z
+echo "Please depress Z probe for 1s"
+while sensors.probes[0].value[0] == 0
+  G4 P500
+echo "Z probe test passed"
+
+; U
+echo "Rotate U axis to minimum"
+M577 U S1
+echo "Rotate U Axis to maximum (*insert spring stretch explanation)"
```

Comment:
This gave me a good chuckle. I can fill this in later.

Reply:
In the meantime, we can probably just say:
"Rotate U axis in the opposite direction until you hear it click the internal switch. (maximum travel)"